program/src/processor.rs

```diff
@@ -464,6 +463,11 @@ fn process_harvest_rewards(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
     **holder_rewards_pool_info.try_borrow_mut_lamports()? = new_holder_rewards_pool_lamports;
     **token_account_info.try_borrow_mut_lamports()? = new_token_account_lamports;
 
+    // Update the holder rewards state with the new "unharvested" rewards.
+    holder_rewards_state.unharvested_rewards = holder_rewards_state
+        .unharvested_rewards
+        .saturating_sub(rewards_to_harvest);
```

Comment:
This is an important case to consider, and I think we should keep it simple -- if there's not enough rewards in the pool because of burned tokens, too bad, you still get your unharvested rewards set to 0. Otherwise, I worry we get into situations where people always have some amount of unharvested rewards and get confused. Or worse, there's a constant race between these users to recoup their loss.

On the flipside, any deficits in the rewards pool could be fixed by transferring lamports in directly, so holders could eventually be made whole. We've leaned towards simplicity with this project, so I lean towards just setting it to 0. What do you think?